src/bridge/cockpitchannel.c

```diff
@@ -1424,3 +1451,24 @@ cockpit_channel_parse_stream (CockpitChannel *self)
 
   return connectable;
 }
+
+void
+cockpit_channel_set_group_object (CockpitChannel *self,
+                                  const gchar *key,
+                                  gpointer object)
+{
+  if (self->priv->group_objects == NULL)
+    self->priv->group_objects = g_hash_table_new_full (g_str_hash, g_str_equal, g_free, g_object_unref);
+
+  g_hash_table_replace (self->priv->group_objects, g_strdup (key), g_object_ref (object));
```

Comment:
Note to self: `g_value_dup_boxed()` essentially translates to `g_hash_table_ref()` here, so all is well.